examples/01-transient_analyses/03-mode_superposition_transient.py

```diff
@@ -0,0 +1,140 @@
+# Modal superposition on all modes available
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Create a datasource with upstream modal information
+data_sources = dpf.DataSources(msup_transient_files["rdsp"])
+
+up_stream_data_sources = dpf.DataSources(msup_transient_files["mode"])
+up_stream_data_sources.add_file_path(msup_transient_files["modal_rst"])
+data_sources.add_upstream(up_stream_data_sources)
+
+# Load into a model
+model = dpf.Model(data_sources)
+print(model)
+
+###############################################################################
+# Get the expanded displacement fields for each time-step
+disp = model.results.displacement.on_all_time_freqs.eval()
+print(disp)
+
+###############################################################################
+# Animate the result
+disp.animate(scale_factor=5.0)
+
+###############################################################################
+# Get the expanded displacement fields on selected time-steps:
+disp = model.results.displacement.on_time_scoping([1, 2, 3]).eval()
+print(disp)
+
+###############################################################################
+# Get the expanded displacement fields on part of the mesh, for selected time-steps:
+# TODO: NOT WORKING
+# partial_scoping = dpf.mesh_scoping_factory.nodal_scoping(
+#     model.metadata.meshed_region.nodes.scoping.ids[:200]
+# )
+# disp = model.results.displacement.on_time_scoping(
+#     [1, 2, 3]
+# ).on_mesh_scoping(
+#     partial_scoping
+# ).eval()
+# print(disp)
+
+###############################################################################
+# Modal superposition on selected modes
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# To select a subset of modes for expansion, you cannot use the source operators directly.
+# Below is a workflow to extract results while specifying modes for expansion:
+
+# First build a data source for the modal response factors
+transient_response_ds = dpf.DataSources(result_path=msup_transient_files["rdsp"])
+
+# Define the time-steps of interest
+time_scoping = dpf.time_freq_scoping_factory.scoping_by_sets(list(range(1, 22)))
+
+# Extract the result of interest
+displacement_fc = dpf.operators.result.displacement(
+    data_sources=transient_response_ds,  # Input here the modal response data source
+    time_scoping=time_scoping,  # Input here the time-steps of interest
+    mesh_scoping=None  # Specify here the region of interest
+).eval()
+
+# The FieldsContainer contains one field per time-step, with data for each mode (entity)
+print(displacement_fc)
+
+###############################################################################
+# Now scope the FieldsContainer to the modes of interest
+mode_scoping = dpf.Scoping(ids=list(range(1, 3)))  # Modes of interest
+displacement_fc = dpf.operators.scoping.rescope_fc(
+    fields_container=displacement_fc,
+    mesh_scoping=mode_scoping,  # Input here the modes of interest
```

Comment:
If you input this to the `mesh_scoping`, it is understood as a Nodal scoping. Is this what you mean?

Reply:
@rafacanton no, interestingly enough, this displacement_fc contains fields located at `Modal`, which seems to be a specificity of using an `rdsp` file as input, hence the rescope on a generic Scoping (tried instantiating a Scoping as `Modal` but it does not work):
```
DPF  Fields Container
  with 21 field(s)
  defined on labels: time 

  with:
  - field 0 {time:  1} with Modal location, 1 components and 6 entities.
  - field 1 {time:  2} with Modal location, 1 components and 6 entities.
  - field 2 {time:  3} with Modal location, 1 components and 6 entities.
  - field 3 {time:  4} with Modal location, 1 components and 6 entities.
  - field 4 {time:  5} with Modal location, 1 components and 6 entities.
  - field 5 {time:  6} with Modal location, 1 components and 6 entities.
  - field 6 {time:  7} with Modal location, 1 components and 6 entities.
  - field 7 {time:  8} with Modal location, 1 components and 6 entities.
  - field 8 {time:  9} with Modal location, 1 components and 6 entities.
  - field 9 {time:  10} with Modal location, 1 components and 6 entities.
  - field 10 {time:  11} with Modal location, 1 components and 6 entities.
  - field 11 {time:  12} with Modal location, 1 components and 6 entities.
  - field 12 {time:  13} with Modal location, 1 components and 6 entities.
  - field 13 {time:  14} with Modal location, 1 components and 6 entities.
  - field 14 {time:  15} with Modal location, 1 components and 6 entities.
  - field 15 {time:  16} with Modal location, 1 components and 6 entities.
  - field 16 {time:  17} with Modal location, 1 components and 6 entities.
  - field 17 {time:  18} with Modal location, 1 components and 6 entities.
  - field 18 {time:  19} with Modal location, 1 components and 6 entities.
  - field 19 {time:  20} with Modal location, 1 components and 6 entities.
  - field 20 {time:  21} with Modal location, 1 components and 6 entities.
```
While trying out different combinations, I realized that this is really not straightforward with the API currently in place. For example, the `Modal` scoping seems to exist server-side but is impossible to build client-side.

Reply:
Conclusion: this example is useful to showcase how to do this, but really we lack helpers regarding mode manipulation.

Reply:
Yes, it's a bit confusing, I think it's worth adding a comment to explain